### PR TITLE
(4.0) Re-implement GodotPayment Android plugin using the Google Play Billing library

### PR DESCRIPTION
**Pull request by timoschwarzer**
This is the same as https://github.com/godotengine/godot/pull/39034 but for 4.0

### CODE REVIEW

File: platform/android/java/plugins/godotpayment/src/main/java/org/godotengine/godot/plugin/payment/GodotPayment.java [hunk 1]
```diff
@@ -32,214 +32,178 @@
...
 
-import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Set;
 
-import org.json.JSONException;
-import org.json.JSONObject;
+public class GodotPayment extends GodotPlugin implements PurchasesUpdatedListener, BillingClientStateListener {
```
Comment (akien-mga) [inline platform/android/java/plugins/godotpayment/src/main/java/org/godotengine/godot/plugin/payment/GodotPayment.java:63]: This empty line should be removed to match the new code style.

Comment (timoschwarzer) [reply platform/android/java/plugins/godotpayment/src/main/java/org/godotengine/godot/plugin/payment/GodotPayment.java]: Sorry, forgot to run `clang-format` again... fixed :sweat_smile:

Comment (akien-mga) [reply platform/android/java/plugins/godotpayment/src/main/java/org/godotengine/godot/plugin/payment/GodotPayment.java]: You can install pre-commit hooks that do it for you: https://github.com/godotengine/godot/tree/master/misc/hooks

Comment (timoschwarzer) [reply platform/android/java/plugins/godotpayment/src/main/java/org/godotengine/godot/plugin/payment/GodotPayment.java]: Thanks! I already did that but I have two separate clones for 3.2 and 4.0 and forgot to add it in my 4.0 directory 😅

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**